client/marklogicClient.ts

```diff
@@ -474,9 +489,12 @@ export function requestMarkLogicUnitTest(
     );
 }
 
-export function newMarklogicManageClient(dbClientContext: ClientContext, managePort: number): ClientContext {
+export function newMarklogicClientWithPort(
```

Comment:
Something to consider if you may need to add a 3rd argument here in the future - allow the caller to pass in an `overrides` object that lets the caller override whatever is in the `dbClientContext`. That means when a 3rd arg is needed, you don't have to change the method signature.

Example of a caller:

    const client = newMarkLogicClient(this.dbClientContext, {"port": somePort, "basePath": somePath});

And then the function would use those overrides to modify the clientParams.. 

The only downside would be that the callers know the names of "port" and "basePath", but that seems reasonable since those are defined in an interface.